pkg/datamover/utils.go

```diff
@@ -0,0 +1,186 @@
+	metadataCacheSize := kopia.GetDataStoreGeneralMetadataCacheSize(kp.Credential.KopiaServerSecret.ConnectOptions)
+	return repository.ConnectToAPIServer(
+		ctx,
+		kp.Credential.KopiaServerSecret.Cert,
+		kp.Credential.KopiaServerSecret.Password,
+		kp.Credential.KopiaServerSecret.Hostname,
+		kp.Location.Endpoint,
+		kp.Credential.KopiaServerSecret.Username,
+		contentCacheSize,
+		metadataCacheSize,
+	)
+}
+
+// connectToKopiaRepositoryServer connects to the kopia server with given repository server CR
+func connectToKopiaRepositoryServer(ctx context.Context, rs *param.RepositoryServer) (string, error) {
+	contentCacheMB, metadataCacheMB := kopiacmd.GetCacheSizeSettingsForSnapshot()
+	hostname, userPassphrase, certData, err := secretsFromRepositoryServerCR(rs)
+	if err != nil {
+		return "", errors.Wrap(err, "Error Retrieving Connection Data from Repository Server")
+	}
+	return userPassphrase, repository.ConnectToAPIServer(
+		ctx,
+		certData,
+		userPassphrase,
+		hostname,
+		rs.Address,
+		rs.Username,
+		contentCacheMB,
+		metadataCacheMB,
+	)
+}
+
+func secretsFromRepositoryServerCR(rs *param.RepositoryServer) (string, string, string, error) {
+	userCredJSON, err := json.Marshal(rs.Credentials.ServerUserAccess.Data)
+	if err != nil {
+		return "", "", "", errors.Wrap(err, "Error Unmarshalling User Credentials")
+	}
+	certJSON, err := json.Marshal(rs.Credentials.ServerTLS.Data)
+	if err != nil {
+		return "", "", "", errors.Wrap(err, "Error Unmarshalling Certificate")
+	}
+	hostname, userPassphrase, err := hostNameAndUserPassPhraseFromRepoServer(string(userCredJSON))
+	if err != nil {
+		return "", "", "", errors.Wrap(err, "Error Getting Hostname/User Passphrase from User credentials")
+	}
+	return hostname, userPassphrase, string(certJSON), err
+}
+
+func hostNameAndUserPassPhraseFromRepoServer(userCreds string) (string, string, error) {
+	var userAccessMap map[string]string
+	if err := json.Unmarshal([]byte(userCreds), &userAccessMap); err != nil {
+		return "", "", errors.Wrap(err, "Failed to unmarshal User Credentials Data")
+	}
+
+	var userPassPhrase string
+	var hostName string
+	for key, val := range userAccessMap {
+		hostName = key
+		userPassPhrase = val
+	}
+	decodedUserPassPhrase, err := base64.StdEncoding.DecodeString(userPassPhrase)
+	if err != nil {
+		return "", "", errors.Wrap(err, "Failed to Decode User Passphrase")
+	}
+	return hostName, string(decodedUserPassPhrase), nil
+}
+
+// kopiaLocationPush pushes the data from the source using a kopia snapshot
+func kopiaLocationPush(ctx context.Context, path, outputName, sourcePath, password string) error {
+	var snapInfo *snapshot.SnapshotInfo
+	var err error
+	switch sourcePath {
+	case usePipeParam:
+		snapInfo, err = snapshot.Write(ctx, os.Stdin, path, password)
+		log.Print("---- PipeParam ----", field.M{
+			"Source Path":      os.Stdin,
+			"Destination Path": path,
+		})
```

Comment:
Missed cleanup? If you want to log, remove `----`, lets use plaintext.

Reply:
Will remove it